File: LeetCode Practice Problems/206. Reverse Linked List.cpp
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        if(head == NULL) return head;
        ListNode* temp = new ListNode(head->val);
        head = head->next;
        while(head != NULL)
        {
            ListNode* t = new ListNode(head->val);
            t->next = temp;
            temp = t;
            head = head->next;
        }
        
        return temp;
    }
};

File: LeetCode/LeetCode Practice Problems/206-Reverse Linked List.java
class Solution {
    public ListNode reverseList(ListNode head) {
        ListNode rev = null;
        while(head != null) {
            ListNode cur = new ListNode(head.val);
            cur.next = rev;
            rev = cur;
            head = head.next;
        }
        return rev;
    }
}
